homeassistant/components/homewizard_energy/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""The Homewizard Energy integration."""
+import asyncio
+import logging
+
+import aiohwenergy
+import async_timeout
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_API, COORDINATOR, DOMAIN, PLATFORMS
+from .coordinator import HWEnergyDeviceUpdateCoordinator as Coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Homewizard Energy from a config entry."""
+
+    _LOGGER.debug("__init__ async_setup_entry")
+
+    # Get api and do a initialization
+    energy_api = aiohwenergy.HomeWizardEnergy(entry.data.get(CONF_IP_ADDRESS))
+
+    # Validate connection
+    initialized = False
+    try:
+        with async_timeout.timeout(10):
+            await energy_api.initialize()
```

Comment:
I would move this into the coordinator, and then you can let `await coordinator.async_config_entry_first_refresh()` do it's magic. Create `energy_api` object inside the coordinator update method and initialize when created.

Reply:
Moved init to coordinator (https://github.com/home-assistant/core/pull/55812/commits/3e3fb381b8bf9472f84a236ac6e5557694611407)